### PR TITLE
[sirius] Add 1-st implementation of sirius scf check

### PR DESCRIPTION
**Pull request by toxa81**
Tested on UENV Sirius spack stack on A100.

### CODE REVIEW

File: checks/apps/sirius/sirius_check.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+class sirius_scf_base_test(rfm.RunOnlyRegressionTest):
+    test_folder = parameter(['Si63Ge'])
+    executable = 'sirius.scf'
+    strict_check = False
+    use_multithreading = False
+    maintainers = ['antonk']
+    data_ref = 'output_ref.json'
+    fout = 'output.json'
+
+    def __init__(self, num_tasks):
```
Comment (teojgo) [inline checks/apps/sirius/sirius_check.py:24]: The init is not needed, you can set directly the:
`num_tasks` and `executable_opts` at the child class

File: checks/apps/sirius/sirius_check.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+@rfm.simple_test
+class UENV_sirius_scf_check(sirius_scf_base_test,
+                            ExtraLauncherOptionsMixin,
+                            CudaVisibleDevicesAllMixin):
+    valid_systems = ['+uenv -amdgpu']
+    valid_prog_environs = ['+sirius +mpi +openmp']
+    tags = {'parallel_k'}
+    def __init__(self):
+        super().__init__(4)
```
Comment (teojgo) [inline checks/apps/sirius/sirius_check.py:147]: Same here, you can set the `num_tasks` and `executable_opts` at the child class level

**Review by teojgo — CHANGES_REQUESTED** [review]

File: checks/apps/sirius/sirius_check.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
+import json
+import pathlib
+import re
```
Comment (teojgo) [inline checks/apps/sirius/sirius_check.py:3]: This is not needed

File: checks/apps/sirius/sirius_check.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+
+@rfm.simple_test
+class UENV_sirius_scf_check(sirius_scf_base_test,
+                            ExtraLauncherOptionsMixin,
+                            CudaVisibleDevicesAllMixin):
+    valid_systems = ['+uenv -amdgpu']
+    valid_prog_environs = ['+sirius +mpi +openmp']
+    tags = {'parallel_k'}
+
```
Comment (teojgo) [inline checks/apps/sirius/sirius_check.py:141]: Delete this line

**Review by teojgo — CHANGES_REQUESTED** [review]

File: checks/apps/sirius/sirius_check.py (hunk 1 above)
Comment (toxa81) [reply checks/apps/sirius/sirius_check.py]: fixed

File: checks/apps/sirius/sirius_check.py (hunk 2 above)
Comment (toxa81) [reply checks/apps/sirius/sirius_check.py]: fixed

File: checks/apps/sirius/sirius_check.py (hunk 3 above)
Comment (toxa81) [reply checks/apps/sirius/sirius_check.py]: fixed

File: checks/apps/sirius/sirius_check.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    @run_after('init')
+    def setup_test(self):
+        self.descr = 'Sirius SCF check'
+        #self.env_vars = {
+        #    'MPICH_OFI_STARTUP_CONNECT': 1,
+        #    'OMP_PLACES': 'cores',
+        #    'OMP_PROC_BIND': 'close'
+        #}
+
+        if self.current_system.name in {'hohgant'}:
```
Comment (teojgo) [inline checks/apps/sirius/sirius_check.py:33]: This is the final comment that I have. It's better to rely on the topology of the node and not on the actual system name. Something similar to this: https://github.com/eth-cscs/cscs-reframe-tests/blob/7100e36f86d72ca93fcaaf31c5c2edc49a5bbdf8/checks/microbenchmarks/cpu/stream/stream.py#L70

**Review by teojgo — CHANGES_REQUESTED** [review]

**Comment by toxa81** [comment]
@teojgo @ekouts Can you  please have a look? Is it good enough for 1st implementation?

**Review by ekouts — APPROVED** [review]

**Review by teojgo — CHANGES_REQUESTED** [review]
Use the clariden-uenv configuration that is now merged in the alps branch.

**Review by teojgo — CHANGES_REQUESTED** [review]
I think you should also remove the tags from the tests.

**Comment by toxa81** [comment]
@teojgo I removed hohgant config, and also tag in the test.

**Review by teojgo — APPROVED** [review]

**Pull request merged**